mongolog/handlers.py

```diff
@@ -231,14 +154,40 @@ def check_keys(self, record):
             return record
 
         for k, v in record['msg'].items():
-            record['msg'][self.new_key(k)] = record['msg'].pop(k)
-
-            if isinstance(v, dict):
-                for old_key in record['msg'][k].keys():
-                    record['msg'][k][self.new_key(old_key)] = record['msg'][k].pop(old_key)
+            self._check_keys(k, v, record['msg'])
 
         return record
 
+    def _check_keys(self, k, v, _dict):
+        """
+        Recursively check key's for special characters.
+        """
+        _dict[self.new_key(k)] = _dict.pop(k)
+
+        if isinstance(v, dict):
+            for nk, vk in v.items():
+                self._check_keys(nk, vk, v)
+
+    def new_key(self, key):
+        """
+        mongo < 3.6
+            Repalce . and $ with Unicode full width equivalents
+        mongo >= 3.6
+            As of mongo 3.6 . and $ are permitted in keys.  But keys may not start with $
+            If we encounter a key that starts with a $ we replace it with it's unicode full width equivalent.
+        """
+        if self.mongo_version >= 3.6:
+            if key[0] == "$":
+                key = u"＄" + key[1:]
+        else:
+            # Older mongo doesn't support $ or . in keys
+            if "." in key:
+                key = key.replace(u".", u"．")
```

Comment:
Crazy, I've never seen `u"．"` before.

Reply:
These are the full width unicode equivalents... This was a recommended approach.  Normally I'd rather through and error to let user know that there is an invalid key... but when it comes to logging you can't really afford to ever let it fail so the special characters are swapped out for the full width equivalents.